### PR TITLE
Support for Rails 5

### PR DESCRIPTION
**Pull request by elektronaut**
Hi! I'd really love to have this gem updated for Rails 5.

I've taken the work done by @schuylr in #32, rebased onto master and fixed the tests so that they're running on the final 5.0 release without deprecations.

This bumps the version to 2.0.0 and drops support for Rails 4.

### CODE REVIEW

**Comment by cguyer** [comment]
Can we get this merged in. Also @elektronaut happen to have a fork of this public till merge?

**Comment by elektronaut** [comment]
@cguyer sure, I've been using this:

```ruby
gem "actionpack-page_caching",
    git: "https://github.com/kord-as/actionpack-page_caching",
    branch: "rails5"
```

**Comment by RogerE** [comment]
There does not seem to be much focus on this gem from Rails core. Understandable of course, they probably don't use it and have lots if other things to deal with. Maybe it could be spun out from the Rails umbrella like acts_as_list and will_paginate?

**Comment by elektronaut** [comment]
The only real change here is renaming an `after_filter` callback to `after_action`, the rest is just test instrumentation and version housekeeping. Please let me know if there's anything else I can do to facilitate a release.

FWIW, I've still got several projects depending on this, and I'm willing to throw my hat in the ring and maintain it as-is if the Rails core team wants to abandon it.

File: Gemfile [hunk 1]
```diff
@@ -2,5 +2,5 @@ source 'https://rubygems.org'
 
 gemspec
 
-gem 'rails'
+gem 'rails', '>= 5.0.0beta1.0'
```
Comment (rafaelfranca) [inline Gemfile]: No need of this.

File: .travis.yml [hunk 2]
```diff
@@ -2,19 +2,13 @@ language: ruby
...
 rvm:
-  - 1.9.3
-  - 2.0.0
-  - 2.1
+  - 2.2.4
   - 2.3.1
 gemfile:
   - Gemfile
-  - gemfiles/Gemfile-4-0-stable
-  - gemfiles/Gemfile-4-1-stable
```
Comment (rafaelfranca) [inline .travis.yml]: We should keep this if we are not dropping support to it

File: .travis.yml [hunk 3]
```diff
@@ -2,19 +2,13 @@ language: ruby
 before_install:
   - gem install bundler
 rvm:
-  - 1.9.3
```
Comment (rafaelfranca) [inline .travis.yml]: Why those ruby versions are being removed?

**Comment by schuylr** [comment]
@rafaelfranca I noticed your diff comments which are mostly on my legwork - unfortunately we can't support both Rails 4.x and Rails 5 in the same codebase. Subsequently, Rails 5 doesn't support the older Ruby versions that you're inquiring as to why they're being removed.

See these comments:

https://github.com/rails/actionpack-page_caching/issues/31#issuecomment-175094969
https://github.com/rails/actionpack-page_caching/pull/32#issuecomment-177225907

I would follow @eileencodes's suggested strategy, and derive two supported branches with a bump to v2 for this gem.

**Comment by rafaelfranca** [comment]
I think we can support Rails 4 and older versions of ruby. There is no changes in the code base only in the test suite. In that case is possible to only run Rails 5 with the supported versions and it is possible to conditionally define code depending in the Rails version to make the test suite pass with 4.x.

**Comment by elektronaut** [comment]
I agree, Rails 4.x support is totally possible. 

I've got 4.0, 4.1, 4.2 and 5.0 working locally, let's see what the CI says.

**Comment by elektronaut** [comment]
That should do the trick. I've changed the version number to a point release, since that would be more appropriate.

**Comment by romenigld** [comment]
Thank's elektronaut!! works fine for me: 
```
gem "actionpack-page_caching",
    git: "https://github.com/kord-as/actionpack-page_caching",
    branch: "rails5"
```

**Comment by pixeltrix** [comment]
@elektronaut sorry, I couldn't use this PR but hopefully you'll be happy now that Rails 5.0 support has been officially added - thanks for your contribution.

**Comment by RogerE** [comment]
Will there be a new gem version released for this?

**Comment by pixeltrix** [comment]
@RogerE Yes - I'm also tidying up the action caching gem and will be releasing a new version of that too.

**Comment by RogerE** [comment]
@pixeltrix excellent, thanks!

**Comment by tomhughes** [comment]
Is there any news on getting the new version out?

**Comment by sighmon** [comment]
@tomhughes I've pointed my `Gemfile` directly at this git repo to test it out with Rails 5.

```Ruby
gem 'actionpack-page_caching', git: 'https://github.com/rails/actionpack-page_caching.git'
```

**Comment by pixeltrix** [comment]
@tomhughes I'll be getting releases out for this and action gems out over Christmas

**Comment by enrico** [comment]
@pixeltrix Any chance we will also get a new gem version for rails/rails-observers ?
actionpack-page_caching and rails-observers are the only 2 entries in my Gemfile not requiring a specific version and instead pointing to git directly...

**Comment by pixeltrix** [comment]
@enrico not sure about rails-observers - will have to check what's involved in that

**Comment by onesupercoder** [comment]
Any help needed with this?

**Comment by pixeltrix** [comment]
@onesupercoder actionpack-page_caching or rails-observers? No for the former and yes for the latter :smile:

I'm just writing a blog post for the release of this and the action caching gem - out at the weekend.

**Comment by onesupercoder** [comment]
Awesome!! Thanks a million!  :-)

**Pull request closed without merging**